packages/react-instantsearch-core/src/core/createConnector.tsx

```diff
@@ -349,28 +411,30 @@ export function createConnectorWithoutContext(
   };
 }
 
-const createConnectorWithContext = (connectorDesc: ConnectorDescription) => (
-  Composed: ReactType
-) => {
-  const Connector = createConnectorWithoutContext(connectorDesc)(Composed);
-
-  const ConnectorWrapper: React.FC<any> = props => (
-    <InstantSearchConsumer>
-      {contextValue => (
-        <IndexConsumer>
-          {indexContextValue => (
-            <Connector
-              contextValue={contextValue}
-              indexContextValue={indexContextValue}
-              {...props}
-            />
-          )}
-        </IndexConsumer>
-      )}
-    </InstantSearchConsumer>
-  );
-
-  return ConnectorWrapper;
-};
+export default function createConnectorWithContext<WidgetProps = {}>(
+  connectorDesc: ConnectorDescription<WidgetProps>
+) {
+  return (Composed: ReactType) => {
+    const Connector = createConnectorWithoutContext<WidgetProps>(connectorDesc)(
+      Composed
+    );
+
+    const ConnectorWrapper: React.FC<any> = props => (
```

Comment:
```suggestion
    const ConnectorWrapper: React.FC<WidgetProps> = props => (
```

Reply:
if I do that, I get an error I don't understand:

```
Type '{ contextValue: InstantSearchContext; indexContextValue: IndexContext; } & TWidgetProps & { children?: ReactNode; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<createConnectorWithoutContext<TWidgetProps>.Connector> & Defaultize<MergePropTypes<Readonly<{ children?: ReactNode; }> & Readonly<ConnectedProps<TWidgetProps>>, InferProps<...>>, Partial<...> | undefined>'.
  Type '{ contextValue: InstantSearchContext; indexContextValue: IndexContext; } & TWidgetProps & { children?: ReactNode; }' is not assignable to type 'Defaultize<MergePropTypes<Readonly<{ children?: ReactNode; }> & Readonly<ConnectedProps<TWidgetProps>>, InferProps<WeakValidationMap<TWidgetProps> | undefined>>, Partial<...> | undefined>'
```